metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/common/helpers/MacrosUpdater.java

```diff
@@ -0,0 +1,433 @@
+	 * @param hostname            The remote hostname.
+	 * @return The updated string with replaced macros.
+	 */
+	public static String update(
+		String text,
+		String username,
+		char[] password,
+		String authenticationToken,
+		@NonNull final String hostname
+	) {
+		if (text == null || text.isEmpty()) {
+			return EMPTY;
+		}
+
+		// Null values control
+		final String passwordAsString = password != null ? String.valueOf(password) : EMPTY;
+		username = username != null ? username : EMPTY;
+		authenticationToken = authenticationToken != null ? authenticationToken : EMPTY;
+
+		String updatedContent = text;
+		if (updatedContent.contains("%{") || updatedContent.contains("${")) {
+			final Map<String, String> simpleMacroNameToField = Map.of(
+				USERNAME.name(),
+				username,
+				PASSWORD.name(),
+				passwordAsString,
+				HOSTNAME.name(),
+				hostname,
+				AUTHENTICATIONTOKEN.name(),
+				authenticationToken
+			);
+
+			final Matcher matcher = MACRO_PATTERN.matcher(text);
+
+			while (matcher.find()) {
+				final String escapeType = matcher.group(2);
+				final String macroName = matcher.group(4);
+				updatedContent =
+					processMacro(
+						updatedContent,
+						matcher.group(0),
+						macroName,
+						escapeType,
+						passwordAsString,
+						username,
+						authenticationToken,
+						simpleMacroNameToField
+					);
+			}
+		}
+
+		return updatedContent;
+	}
+
+	/**
+	 * Processes the macro found in the text, replacing it with the corresponding value based on the macro name.
+	 *
+	 * @param content             The content string with macros.
+	 * @param matchedString       The matched macro string.
+	 * @param macroName           The name of the macro to replace.
+	 * @param escapeType          The escape type for the macro value (e.g., JSON, XML).
+	 * @param passwordAsString    The HTTP password as a string.
+	 * @param username            The HTTP username.
+	 * @param authenticationToken  The HTTP authentication token.
+	 * @param macroNameField      A map of macro names and their corresponding values.
+	 * @return The content with the macro replaced by the corresponding value.
+	 */
+	private static String processMacro(
+		final String content,
+		final String matchedString,
+		final String macroName,
+		final String escapeType,
+		final String passwordAsString,
+		final String username,
+		final String authenticationToken,
+		final Map<String, String> macroNameField
+	) {
+		String updatedContent = content;
+		if (macroName.startsWith(PASSWORD_BASE64.name())) {
+			// PasswordBase64 macros replacement
+			updatedContent = replacePasswordBase64(updatedContent, escapeType, matchedString, passwordAsString);
+		} else if (macroName.startsWith(BASIC_AUTH_BASE64.name())) {
+			// BasicAuthBase64 macros replacement
+			updatedContent =
+				replaceBasicAuthBase64MacroValue(updatedContent, escapeType, matchedString, username, passwordAsString);
+		} else if (macroName.startsWith(MacroType.SHA256_AUTH.name())) {
```

Comment:
Other MacroType are imported while SHA256_AUTH is called using `MacroType.SHA256_AUTH`. Import them all or use `MacroType.<macro-name>` everywhere.